src/main/java/com/hubspot/smtp/client/SmtpSession.java

```diff
@@ -99,6 +99,8 @@ public class SmtpSession {
   private volatile boolean requiresRset = false;
   private volatile EhloResponse ehloResponse = EhloResponse.EMPTY;
 
+  private List<String> localExtensionsList;
```

Comment:
I have my doubts over making this localExtensionsList Global and whether it should be kept as a volatile variable or not. Comments are welcome for edits